servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,197 @@
+    READ_ENTITY_VALUE,
+    UPDATE_ENTITY,
+    DELETE_ENTITY;
+  }
+
+  @Inject
+  public CelAccessChecker(QuarkusNessieAuthorizationConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void canCreateReference(AccessContext context, NamedRef ref) throws ForbiddenException {
+    canPerformOpOnReference(context, ref, CREATE_REFERENCE);
+  }
+
+  @Override
+  public void canAssignRefToHash(AccessContext context, NamedRef ref) throws ForbiddenException {
+    canPerformOpOnReference(context, ref, ASSIGN_REFERENCE_TO_HASH);
+  }
+
+  @Override
+  public void canDeleteReference(AccessContext context, NamedRef ref) throws ForbiddenException {
+    canPerformOpOnReference(context, ref, DELETE_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntries(AccessContext context, NamedRef ref) throws ForbiddenException {
+    canPerformOpOnReference(context, ref, READ_ENTRIES);
+  }
+
+  @Override
+  public void canListCommitLog(AccessContext context, NamedRef ref) throws ForbiddenException {
+    canPerformOpOnReference(context, ref, LIST_COMMIT_LOG);
+  }
+
+  @Override
+  public void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws ForbiddenException {
+    canPerformOpOnReference(context, ref, COMMIT_CHANGE_AGAINST_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, READ_ENTITY_VALUE);
+  }
+
+  @Override
+  public void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, UPDATE_ENTITY);
+  }
+
+  @Override
+  public void canDeleteEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, DELETE_ENTITY);
+  }
+
+  private void canPerformOpOnReference(
+      AccessContext context, NamedRef ref, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        config.rules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return CELUtil.SCRIPT_HOST
+                        .buildScript(entry.getValue().replace("\"", ""))
+                        .withContainer(CELUtil.CONTAINER)
+                        .withDeclarations(CELUtil.AUTHORIZATION_RULE_DECLARATIONS)
+                        .build()
+                        .execute(
+                            Boolean.class,
+                            ImmutableMap.of(
+                                "ref",
+                                ref.getName(),
+                                "role",
+                                getRoleName(context),
+                                "op",
+                                type.name()));
+                  } catch (ScriptException e) {
+                    throw new ForbiddenException(
```

Comment:
Is this really a forbidden exception? Or is it a 500 error?

Reply:
imo it should be a `ForbiddenException` since it indicates that access to a resource has been forbidden (thus throwing a HTTP 403). Throwing a 500 error would mean something went wrong on the server-side , which the client usually can't do anything about

Reply:
This call happens when something goes wrong w/ calculating permissions though right? It means that the server rules contain a syntax error or some other server side error. This doesn't mean the user is unauthorised, rather that the server is having an issue.

Reply:
oh you're absolutely right. I think I copy/pasted the wrong exception during refactoring, so this definitely should be a 5xx